src/sentry/tsdb/snuba.py

```diff
@@ -276,8 +335,10 @@ def get_data(
 
         orderby = []
         if group_on_time:
-            orderby.append("-time")
-
+            if not model_requires_manual_group_on_time:
+                orderby.append("-time")
+            else:
+                orderby.append(f"-{group_on_time_column_alias}")
```

Comment:
Should we just alias this column as `time` to avoid some of these extra branches?

Reply:
Yeah, I tried that and somewhere down the processing pipes it undesirably mutates the query if we use `time` as an alias.